Formula/gnat.rb

```diff
@@ -0,0 +1,152 @@
+class Gnat < Formula
+  desc "GNU New York University Ada Trans"
+  homepage "https://gcc.gnu.org/"
+  url "https://ftp.gnu.org/gnu/gcc/gcc-11.1.0/gcc-11.1.0.tar.xz"
+  mirror "https://ftpmirror.gnu.org/gcc/gcc-11.1.0/gcc-11.1.0.tar.xz"
+  sha256 "4c4a6fb8a8396059241c2e674b85b351c26a5d678274007f076957afa1cc9ddf"
+  license "GPL-3.0-or-later" => { with: "GCC-exception-3.1" }
+  head "https://gcc.gnu.org/git/gcc.git"
+
+  livecheck do
+    url "https://ftp.gnu.org/gnu/gcc/gcc-11.1.0"
+    regex(%r{href=.*?gcc[._-]v?(\d+(?:\.\d+)+)(?:/?["' >]|\.t)}i)
+  end
+
+  # The bottles are built on systems with the CLT installed, and do not work
+  # out of the box on Xcode-only systems due to an incorrect sysroot.
+  pour_bottle? do
+    on_macos do
+      reason "The bottle needs the Xcode CLT to be installed."
+      satisfy { MacOS::CLT.installed? }
+    end
+  end
+
+  depends_on arch: :x86_64
+
+  depends_on "gmp"
+  depends_on "isl"
+  depends_on "libmpc"
+  depends_on "mpfr"
+  depends_on "zstd"
+
+  uses_from_macos "libiconv"
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "binutils"
+  end
+
+  conflicts_with "gcc", because: "both install gcc"
+
+  # GCC bootstraps itself, so it is OK to have an incompatible C++ stdlib
+  cxxstdlib_check :skip
+
+  def version_suffix
+    if build.head?
+      "HEAD"
+    else
+      version.major.to_s
+    end
+  end
+
+  resource "bootstrap_gcc" do
+    url "https://phoenixnap.dl.sourceforge.net/project/gnuada/GNAT_GCC%20Mac%20OS%20X/11.1.0/native/gcc-11.1.0-x86_64-apple-darwin15.pkg"
+    sha256 "d947b5db0576cb62942e5ce61f3ef53fb679f07b1adff7a4c0fa19a5e72a9532"
+  end
+
+  def install
+    resource("bootstrap_gcc").stage do
+      system "pkgutil", "--expand-full", "gcc-11.1.0-x86_64-apple-darwin15.pkg", buildpath/"bootstrap_gcc"
+    end
+    bootstrap_gcc_prefix = buildpath/"bootstrap_gcc/gcc-11.1.0-x86_64-apple-darwin15.pkg/Payload"
+    inreplace "configure", /\${CC}(?= -c conftest\.adb)/, bootstrap_gcc_prefix/"bin/gcc"
+    open("gcc/ada/gcc-interface/Make-lang.in", "a") { |f| f.puts "override CC = #{bootstrap_gcc_prefix}/bin/gcc" }
+
+    ENV.append_path "PATH", bootstrap_gcc_prefix/"bin"
+    ENV["ADAC"] = bootstrap_gcc_prefix/"bin/gcc"
+
+    # GCC will suffer build errors if forced to use a particular linker.
+    ENV.delete "LD"
+
+    pkgversion = "Homebrew GCC #{pkg_version} #{build.used_options*" "}".strip
+    cpu = Hardware::CPU.arm? ? "aarch64" : "x86_64"
+
+    args = %W[
+      --prefix=#{prefix}
+      --libdir=#{lib}/gcc/#{version_suffix}
+      --disable-nls
+      --enable-checking=release
+      --enable-languages=c,ada
```

Comment:
Is enabling C necessary to build Ada?

Reply:
Ada depends on C

Reply:
Sure, but that doesn't mean I need to build a C compiler just to build an Ada compiler. Why can't I build an Ada compiler while still using my already existing C compiler?

Reply:
Ada part of `GNAT` need use some gcc's C code.